plugins/kubernetes/app/models/kubernetes/role.rb

```diff
@@ -60,6 +60,20 @@ def self.configured_for_project(project, git_sha)
       end
     end
 
+    def defaults
+      return unless raw_template = project.repository.file_content(config_file, 'HEAD', update: false)
+      objects = Array.wrap(Kubernetes::Util.parse_file(raw_template, config_file))
+      return unless deploy = objects.detect { |o| ['Deployment', 'DaemonSet'].include?(o.fetch('kind')) }
+
+      replicas = deploy['spec']['replicas']
+
+      return unless limits = deploy['spec']['template']['spec']['containers'].first['resources'].try(:[], 'limits')
+      cpu = limits['cpu'].to_i / 1000.0 # 250m -> 0.25
+      ram = limits['ram'].to_i # 200Mi -> 200
```

Comment:
Not accurate.  Look at https://github.com/kubernetes/kubernetes/blob/master/docs/design/resources.md#resource-types